### PR TITLE
Update QUICHE from 96a9f6bd7 to c8c21d0e9

### PR DESCRIPTION
**Pull request by RenjieTang**
https://github.com/google/quiche/compare/96a9f6bd7..c8c21d0e9

$ git log 96a9f6bd7..c8c21d0e9 --date=short --no-merges --format="%ad %al %s"

2021-12-22 quiche-dev Clear the OgHttp2Session pending frames queue on starting an immediate GOAWAY.
2021-12-21 quiche-dev Misc. cleanups: add missing includes, comment corrections, consistent param naming.
2021-12-20 quiche-dev Adds an option for maximum header field size to OgHttp2Adapter.
2021-12-20 quiche-dev Handle false return values from visitor_.OnMetadataEndForStream() in OgHttp2Session.
2021-12-20 quiche-dev Avoid sending frames after sending an immediate GOAWAY in OgHttp2Session.
2021-12-20 quiche-dev Allow OgHttp2Session to blackhole data on connection errors.

Signed-off-by: Renjie Tang <renjietang@google.com>

Risk Level: Low
Testing: Existing tests
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

File: source/common/http/http2/metadata_encoder.cc
Comment (RyanTheOptimist) [inline source/common/http/http2/metadata_encoder.cc:146]: Do you have any idea what implementing this method might entail?

**Review by RyanTheOptimist — DISMISSED** [review]
LGTM, modulo a question about the unimplemented OnFailure method.

File: source/common/http/http2/metadata_encoder.cc
Comment (RenjieTang) [reply source/common/http/http2/metadata_encoder.cc:146]: I'm not quite sure how this method should be implemented here.
The other subclass is SelfDeletingMetadataSource but I think these 2 classes are quite different.
Adding @birenroy @diannahu who might know more about this.
We could wait for an actual impl to merge this PR or we can land first and then implement later. Either way I'm happy.

Comment (RyanTheOptimist) [reply source/common/http/http2/metadata_encoder.cc:146]: Makes sense. I suspect that the current no-op implementation is safe to land but it does make me wonder. How 'bout we wait for either of @birenroy or @diannahu to comment and then land based on their input?

Comment (diannahu) [reply source/common/http/http2/metadata_encoder.cc:146]: Hi, thanks for adding us! I added this interface method for SelfDeletingMetadataSource to know to delete itself on failure. AIUI, SelfDeletingMetadataSource is only used internally to NgHttp2Adapter, so it should be okay for other implementations (including BufferMetadataSource) to leave OnFailure() as a no-op. Thanks!

Comment (RyanTheOptimist) [reply source/common/http/http2/metadata_encoder.cc:146]: Perfect Thanks, Dianna! In that case, @RenjieTang perhaps we should remove the TODO, then we can land it.

**Review by RyanTheOptimist — APPROVED** [review]

**Pull request merged**
